src/components/modal/addtostudio/studio-button.jsx

```diff
@@ -0,0 +1,72 @@
+const truncate = require('lodash.truncate');
+const PropTypes = require('prop-types');
+const React = require('react');
+const classNames = require('classnames');
+const Spinner = require('../../spinner/spinner.jsx');
+
+require('./modal.scss');
+
+const StudioButton = ({
+    hasRequestOutstanding,
+    id,
+    includesProject,
+    title,
+    onToggleStudio
+}) => {
+    const checkmark = (
+        <img
+            alt="checkmark-icon"
+            className="studio-status-icon-checkmark-img"
+            src="/svgs/modal/confirm.svg"
+        />
+    );
+    const plus = (
+        <img
+            alt="plus-icon"
+            className="studio-status-icon-plus-img"
+            src="/svgs/modal/add.svg"
+        />
+    );
+    return (
+        <div
+            className={classNames(
+                'studio-selector-button',
+                {'studio-selector-button-waiting': hasRequestOutstanding},
+                {'studio-selector-button-selected':
+                    includesProject && !hasRequestOutstanding}
+            )}
+            data-id={id}
+            onClick={onToggleStudio}
+        >
+            <div
+                className={classNames(
+                    'studio-selector-button-text',
+                    {'studio-selector-button-text-selected': includesProject},
+                    {'studio-selector-button-text-unselected': !includesProject}
+                )}
+            >
+                {truncate(title, {length: 20, separator: /[.,:;]*\s+/})}
```

Comment:
Aside: This looks like a good candidate for a helper function, since we probably want most things truncated in the same way.

Reply:
I'm putting it in lib/truncate.js